botocore/client.py

```diff
@@ -537,6 +580,10 @@ def config(self):
     def method_to_api_mapping(self):
         return self._method_to_api_mapping
 
+    @property
+    def s3_addressing_style(self):
```

Comment:
Are we planning on exposing  every single config option going forward?  Why not use the config property here?